File: apps/organiclever-be-jasb/src/test/java/com/organiclever/be/integration/steps/HealthSteps.java
package com.organiclever.be.integration.steps;

import com.organiclever.be.integration.ResponseStore;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Scope;
import org.springframework.test.web.servlet.MockMvc;
import org.springframework.test.web.servlet.result.MockMvcResultMatchers;

import static org.assertj.core.api.Assertions.assertThat;
import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;

@Scope("cucumber-glue")
public class HealthSteps {

    @Autowired
    private MockMvc mockMvc;

    @Autowired
    private ResponseStore responseStore;

    @When("^an operations engineer sends GET /health$")
    public void anOperationsEngineerSendsGetHealth() throws Exception {
        responseStore.setResult(mockMvc.perform(get("/health")).andReturn());
    }

    @When("^an unauthenticated engineer sends GET /health$")
    public void anUnauthenticatedEngineerSendsGetHealth() throws Exception {
        responseStore.setResult(mockMvc.perform(get("/health")).andReturn());
    }

    @When("^a client sends GET /health$")
    public void aClientSendsGetHealth() throws Exception {
        responseStore.setResult(mockMvc.perform(get("/health")).andReturn());
    }

    @Then("the health status should be {string}")
    public void theHealthStatusShouldBe(final String expectedStatus) throws Exception {
        MockMvcResultMatchers.jsonPath("$.status").value(expectedStatus)
            .match(responseStore.getResult());
    }

    @Then("the response should not include detailed component health information")
    public void theResponseShouldNotIncludeComponentDetails() throws Exception {
        final String body = responseStore.getResult().getResponse().getContentAsString();
        assertThat(body).doesNotContain("components");
    }
}

File: apps/organiclever-be-e2e/steps/health.steps.ts
import { expect } from "@playwright/test";
import { createBdd } from "playwright-bdd";
import { setResponse, getResponse } from "../utils/response-store";

const { When, Then } = createBdd();

When(/^an operations engineer sends GET \/health$/, async ({ request }) => {
  setResponse(await request.get("/api/v1/health"));
});

When(/^an unauthenticated engineer sends GET \/health$/, async ({ request }) => {
  setResponse(await request.get("/api/v1/health"));
});

// oxlint-disable-next-line no-empty-pattern
Then("the health status should be {string}", async ({}, status: string) => {
  const body = (await getResponse().json()) as Record<string, unknown>;
  expect(body["status"]).toBe(status);
});

Then("the response should not include detailed component health information", async () => {
  const body = (await getResponse().json()) as Record<string, unknown>;
  expect(body["components"]).toBeUndefined();
});
